### PR TITLE
bzl: extract workspace dependencies to macros

### PR DESCRIPTION
**Pull request by mikedanese**
and name the kubernetes workspace. this makes depending on the core repo
from other bazel workspaces a bit easier as it reduces the copypaste of
WORKSPACE.

```release-note
NONE
```

### CODE REVIEW

File: build/root/WORKSPACE [hunk 1]
```diff
@@ -1,82 +1,19 @@
...
-    urls = ["https://github.com/coreos/etcd/releases/download/v%s/etcd-v%s-linux-amd64.tar.gz" % (ETCD_VERSION, ETCD_VERSION)],
-)
-
-http_archive(
-    name = "io_bazel_rules_docker",
-    sha256 = "c440717ee9b1b2f4a1e9bf5622539feb5aef9db83fc1fa1517818f13c041b0be",
-    strip_prefix = "rules_docker-8bbe2a8abd382641e65ff7127a3700a8530f02ce",
-    urls = ["https://github.com/bazelbuild/rules_docker/archive/8bbe2a8abd382641e65ff7127a3700a8530f02ce.tar.gz"],
-)
+workspace(name = "io_kubernetes")
```
Comment (ixdy) [inline build/root/WORKSPACE]: part of me feels like to match go, this should be "io_k8s_kubernetes" or something, though that's also kinda gross.

File: build/root/WORKSPACE [hunk 2]
```diff
@@ -1,82 +1,19 @@
...
+load("//build:workspace.bzl", "kubernetes_dependencies", "kubernetes_docker_dependencies")
 
-versions.check(minimum_bazel_version = "0.10.0")
+kubernetes_dependencies()
 
-load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains", "go_download_sdk")
-load("@io_bazel_rules_docker//docker:docker.bzl", "docker_repositories", "docker_pull")
+load("@io_bazel_rules_go//go:def.bzl", "go_rules_dependencies", "go_register_toolchains")
 
 go_rules_dependencies()
```
Comment (ixdy) [inline build/root/WORKSPACE]: any reason not to include this in kubernetes_dependencies() too?

File: build/root/WORKSPACE [hunk 3]
```diff
@@ -1,82 +1,19 @@
...
+docker_repositories()
 
-docker_pull(
-    name = "official_busybox",
-    digest = "sha256:4cee1979ba0bf7db9fc5d28fb7b798ca69ae95a47c5fecf46327720df4ff352d",
-    registry = "index.docker.io",
-    repository = "library/busybox",
-    tag = "latest",  # ignored, but kept here for documentation
-)
+kubernetes_docker_dependencies(docker_pull)
```
Comment (ixdy) [inline build/root/WORKSPACE]: why not just load docker_pull in workspace.bzl?

File: build/root/WORKSPACE (hunk 1 above)
Comment (mikedanese) [reply build/root/WORKSPACE]: I'm fine with that if we follow it across k8s.io repos.

Comment (ixdy) [reply build/root/WORKSPACE]: this was more an idle comment instead of really having a strong opinion.

File: build/root/WORKSPACE (hunk 2 above)
Comment (ixdy) [reply build/root/WORKSPACE]: I am curious about this one though.

**Pull request closed without merging**